Fixes areas blobs shouldn't win from spreading to.

ktccd opened the pull request:
:cl: ktccd
bugfix: Blob pieces on some areas no longer count for blob mode victory (Such as space or asteroid).
bugfix: Blobcode no longer spawns and destroys blobs uselessly, which fixes other stuff too.
bugfix: Asteroid areas are no longer valid gang territory.
/:cl:
Fixes  #13126

ktccd committed:
Fixes issue #13126and also a few other areas blobs shouldn't be win from absorbing (centcom, SPACE etc).

:cl: ktccd
Fixes bug #13126 and also a few other areas blobs shouldn't be win from
absorbing (centcom, SPACE etc).
Added a variable to areas.
Kept the original blobs list as a list of all blobs, in case that was
used somewhere else, and instead made a separate list for legit blobs.
/:cl:

ktccd commented:
I think I could expand this to include a similar gang fix to cover issue  #13126 completely, but I'm unsure if I should make a separate PR for that?

Anonus commented:
You need to do `bugfix: Text` or it won't work. Also that's a bad way to phrase the changelog anyway, it should be more like `bugfix: Blob pieces on space tiles or on the mining asteroid no longer count for blob mode victory.`

MrStonedOne commented on code/game/area/Space Station 13 areas.dm:
```diff
@@ -164,6 +165,7 @@ var/list/teleportlocs = list()
 	icon_state = "centcom"
 	requires_power = 0
 	has_gravity = 1
+	blob_legit = 0 //Should go without saying, no blobs should take over centcom as a win condition.
```
This is a child of space and so it wouldn't need this, it would inherit it from space

optimumtact commented:
When in doubt, always err on the side of a seperate PR

ktccd committed:
Fixes issue #13126 (Better this time)

:cl: ktccd
bugfix: Blob pieces on some areas no longer count for blob mode victory
(Such as space or asteroid).
bugfix: Asteroid areas are no longer valid gang territory.
/:cl:

ktccd commented:
Ok, so I fixed the whole #13126 issue, seems asteroid areas were considered valid territory by default (Reason it never broke on boxstation before was because asteroid was ALSO on another z-level. Not the case on some maps).
I also moved my blob variable to the other area file so it's more consistent with how others coded their variables.

optimumtact commented:
If you include the words
Fixes #issuenumber
In your original pr comment (you can edit it)

they will be autoclosed when the pr is merged.

ktccd commented:
Like so?

duncathan commented:
Yep

ktccd commented:
Sweet, thanks :P.

Aranclanos commented on code/game/gamemodes/blob/theblob.dm:
```diff
@@ -19,7 +19,10 @@
 
 /obj/effect/blob/New(loc)
-	blobs += src
+	var/area/Ablob = loc
```
wouldn't the loc be a turf?
use get_area or however that proc is called

ktccd committed:
Fixed looking up the /area of the loc, not the turf.

I messed up, thanks Aranclanos!

ktccd commented:
Alright, I fucked up before in how loc worked. Hopefully should be fixed for real now.

Aranclanos commented:
did you test this?

ktccd commented:
I had not, I just learned how to do so, so I already found a problem with this ^^.
No merging yet guys, I'll test so it works first ^^.

ktccd committed:
Rewrote a bit of blob code to fix... a lot

Stuff fixed: My sad attempt at fixing #13126.
The expand() proc for blobs is hopefully less intensive now (WHY did it
used to spawn a blob only to destroy it? That must've been such a
waste...).
Destroy code no longer needs to adjust for the expand() proc doing weird
things.
The antag panel will now show the correct variable used to determine
blob progress.
Renamed the variables I used so they're not so similar to each other.
blob_legit and blobs_legit was WAY to easy to mix up.

Anonus commented on code/game/gamemodes/blob/theblob.dm:
```diff
@@ -124,24 +128,33 @@
+			B.update_icon()
+		else
+			T.blob_act() //If we cant move in hit the turf
+			qdel(B) //We should never get to this point, since we checked before moving in. Destroy blob anyway for cleanliness though
```
comment is unneeded, as you want the blob to be deleted as it'd otherwise stack up on the expanding blob's turf.

ktccd replied:
Wait, which comment? The one on qdel at the end?
I figured it shouldn't ever get to that else statement, since if there's nothing dense in the way, it should be able to move in?
If there IS another case (is there something else that stops it?), I would like to know, please, as I would like to put a check for it first ^^.

Anonus replied:
There are like one or two cases where it'd fail but I think I might have accidentally fixed those when I changed blob_act().
Basically border objects are terrible and cause issues.

ktccd replied:
border objects? like... the border of Z levels?

Anonus replied:
Stuff like the old windows and firelocks that didn't take up the whole tile but still prevented movement.
We still have windoors, which are also border objects.

ktccd replied:
oh god... ok, so I guess it was a good idea to keep a catch-all conditional...

ktccd committed:
Update to bring code in line with standards

Removes some unecessary text as well.

optimumtact commented on code/game/gamemodes/blob/theblob.dm:
```diff
@@ -124,24 +128,33 @@
 
 	if(!T)	return 0
-	var/obj/effect/blob/B = new /obj/effect/blob/normal(src.loc)
+	//We can keep the above code in this proc, it probably works fine?
```
generally we avoid making too many comments like this one and the one below it.

As a rule of thumb, if the comment doesn't aid in understanding the code then don't bother with it.

ktccd replied:
Alright, noted ^^.

ktccd committed:
Brings code up to current specs

Also removes some of my unneded comments.

ktccd commented:
There, updated ^^. Hope it's more up to specs now :D
